main: added replication controller namespace support

darkcrux opened the pull request:
Currently krud only works on the default namespace. Some replication
controllers may exist on other namespaces. This fix adds support for
defining a namespace for a replication controller.

darkcrux committed:
main: added replication controller namespace support

Currently krud only works on the default namespace. Some replication
controllers may exist on other namespaces. This fix adds support for
defining a namespace for a replication controller.

mjibson commented on main.go:
```diff
@@ -24,6 +24,7 @@ func main() {
 	listen := fs.String("listen", ":9500", "")
 	deploymentKey := fs.String("deployment-key", "deployment", "Key to use to differentiate between two different controllers.")
 	controllerName := fs.String("controller-name", "", "Name of the replication controller to update.")
+	namespace := fs.String("namespace", "", "Namespace the replicationController belongs to.")
```
`"namespace", api.NamespaceDefault, ...`

mjibson replied:
replicationController -> replication controller

mjibson replied:
I think `Replication controller namespace.` is a bit better phrasing for the description argument.

darkcrux replied:
silly me.. that looks like code instead of a nice help message. :)

mjibson commented on README.md:
```diff
@@ -27,6 +27,7 @@ Optional:
 - `KRUD_DEPLOYMENT_KEY`: key to use to differentiate between two different controllers; defaults to `deployment`
 - `KRUD_K8S_ENDPOINT`: kubernetes endpoint; defaults to `http://localhost:8080`
 - `KRUD_LISTEN`: listen address; defaults to `:9500`
+- `KRUD_NAMESPACE`: the namespace the replication controller belongs to; defaults to `default`
```
```
`KRUD_NAMESPACE`: namespace of the replication controller; defaults to the kubernetes default
```

darkcrux replied:
much better!

mjibson commented on main.go:
```diff
@@ -230,8 +233,10 @@ func (k *Krud) update(h *Webhook) error {
 	}
-	ns := api.NamespaceDefault
-	rcs := client.ReplicationControllers(ns)
+	if k.Namespace == "" {
```
My other comments have suggested setting the code's default to api.NamespaceDefault. Those changes would make this block unneeded, since the namespace should always be set by this point.

mjibson commented:
I've suggested using api.NamespaceDefault a bit more. However I could easily be convinced your current approach is better.

darkcrux commented:
Yeah I think we should go with `api.NamespaceDefault`. Less blocks. :) Will update the code.

darkcrux committed:
main: Updated namespace param default

The default namespace is now set to `api.NamespaceDefault` on the arg
part. This eliminates the conditional block that sets the default when
no param is passed.

darkcrux commented:
Updated it with the suggestions. :smile:
